pkg/exasol-rest-api/query.go

```diff
@@ -0,0 +1,53 @@
+package exasol_rest_api
+
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type Application struct {
+	Properties *ApplicationProperties
+}
+
+func (application *Application) Query(context *gin.Context) {
+	response, err := application.queryExasol(context.Param("query"))
+	if err != nil {
+		context.JSON(http.StatusBadRequest, gin.H{
+			"ErrorCode": "EXA-REST-API-1",
+			"Message":   err.Error(),
+		})
+	} else {
+		context.Data(http.StatusOK, "application/json", response)
+	}
+}
+
+func (application *Application) queryExasol(query string) ([]byte, error) {
+	connection, err := application.openConnection()
+	if err != nil {
+		return nil, err
+	}
+	response, err := connection.executeQuery(query)
+	if err != nil {
+		return nil, err
+	}
+	err = connection.close()
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+func (application *Application) openConnection() (*websocketConnection, error) {
+	connection := &websocketConnection{
+		connProperties: application.Properties,
+	}
+	err := connection.connect()
+	if err != nil {
+		return nil, err
+	}
+	err = connection.login()
+	if err != nil {
+		return nil, err
+	}
+	return connection, err
```

Comment:
```suggestion
	return connection, nil
```
`err` must be `nil` here. Explicitly returning `nil` makes this clearer

Reply:
done